### DISCUSSION TITLE
6/30 Discord Meeting

### DISCUSSION THREAD

**Discussion by nibanks** [microsoft/msquic#2855] [Show and tell]
## Agenda

- Discussion of #2071 between @wizmann and @maolson-msft (if both available).
- Plan for the next month's QUIC work:
  - Finish up test work:
    - #2685
    - #2785
    - #2758 
    - #1715
  - Improve VN test coverage:
    - #1218 
    - #1831 
    - #2503
  - Deploy VN & V2 to interop server and interop test runner
  - Start design work on Encryption offload
  - Expose QUIC flow blocked timings via API
  - Improvements to Linux signing and packaging
- Open mic to discuss anything else

**Comment by dahavey** [comment]
Meeting Notes:
06.30.2022 Public MsQuic Discord Meeting

Thursday, June 23, 2022
10:05 AM

	• BBR Congestion Control
		○ What does it take to get this merged?
		○ Open comments(?)
			§ When we are not using BBR this is a lot of mem that is not useful at all in Cubic
				□ And will have a direct impact
			§ Do we care? And is there a way to minimize impact?
			§ For this PR why don't we just clearly mark things that are BBR only?
			§ Need to keep last packet info for every packet?
				□ If we send a batch of 40 packets, do we need to do this 40 times?
					® When you receive an ACK it is for a particular packet and you need it's individual information to calculate the bandwidth, etc.
				□ Need to record the time the last ACKed packet was sent.
				□ When you send this packet we want to know what the rate was at that time?
					® You have to get the ACK of the current packet to know the rate.
				□ Maybe it makes sense to have these in the last packet ACKed struct
		○ Need more code comments.
	• Goal - calculate the rate in between ACKS
		○ Complex to have many sent packets associate with last ACK.
			§ At the start of the flush send what is the current ACK?
	• Let's table this optimization and focus on getting the code merged.
		○ Okay, as long as there is no measurable impact on Cubic
	• Let's resolve all conversations that we can.
	• Take one more pass with an eye on regressions in non-bbr stuff.
		○ Automated tests should resolve this.
	• Test connection close and stream close withing callbacks
		○ Daiki working on this.
	• Test upgrading and downgrading versions behind an LB
	• Design work on encryption offload
		○ Reach out to Intel and Mellanox
			§ Reach out to start design process.
	• QUIC flow block timings behing API
		○ Do this in logging traces right now
		○ Want to add another get-param like the stats
	• Linux signing and packaging
		○ Should automate this
		○ Zip up a file using tar, gz or whatever and sign it with a cert
Tricky part is that there is a script to do the packaging, but it doesn't do the signing part.
